scripts/add_electricity.py

```diff
@@ -433,7 +433,10 @@ def attach_conventional_generators(
         marginal_cost=ppl.marginal_cost,
         capital_cost=ppl.capital_cost,
         build_year=ppl.datein.fillna(0).astype(int),
-        lifetime=(ppl.dateout - ppl.datein).fillna(np.inf),
+        lifetime=(
+            ppl.dateout.replace(" ", "0").apply(lambda x: int(x) if pd.notna(x) else 0)
+            - ppl.datein
+        ),
```

Comment:
Thanks @danielelerede-oet for the proposal.
Could you explain why np.inf does not work?
Filling with 0 may lead to issues with multi-year optimization as zero lifetime should imply the capacity should be dropped.
It may be a better default approach to fill values according to the costs.csv values.
What do you think?